code/weapon/beam.h

```diff
@@ -25,13 +25,14 @@ struct obj_pair;
 struct beam_weapon_info;
 struct vec3d;
 
-// beam types
-#define BEAM_TYPE_A					0				// unidirectional beam
-#define BEAM_TYPE_B					1				// "slash" in one direction
-#define BEAM_TYPE_C					2				// targeting lasers (only lasts one frame)
-#define BEAM_TYPE_D					3				// similar to the type A beams, but takes multiple shots and "chases" fighters around
-#define BEAM_TYPE_E					4				// stupid beam. like type A, only it doesn't aim. it just shoots directly out of the turret
-#define BEAM_TYPE_F					5				// SCP type, highly flexible and configurable
+typedef enum class BeamType {
```

Comment:
These definitions should really include the old letter (and possibly number) used for the beam. It's obvious which is which to us, but in the future it'll be really hard to understand. Additionally, wording like "similar to the type A beams" should also be reworded now